src/Microsoft.CodeAnalysis.Razor/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -136,13 +136,17 @@ private ImmutableArray<IParameterSymbol> GetInvokeMethodParameters(INamedTypeSym
 
             if (methods.Length == 0)
             {
-                throw new InvalidOperationException(
-                    ViewComponentResources.FormatViewComponent_CannotFindMethod(ViewComponentTypes.SyncMethodName, ViewComponentTypes.AsyncMethodName, componentType.ToDisplayString(FullNameTypeDisplayFormat)));
+                throw new InvalidOperationException(ViewComponentResources.FormatViewComponent_CannotFindMethod(
+                    ViewComponentTypes.SyncMethodName,
+                    ViewComponentTypes.AsyncMethodName,
+                    componentType.ToDisplayString(FullNameTypeDisplayFormat)));
             }
             else if (methods.Length > 1)
             {
-                throw new InvalidOperationException(
-                    ViewComponentResources.FormatViewComponent_AmbiguousMethods(componentType.ToDisplayString(FullNameTypeDisplayFormat), ViewComponentTypes.AsyncMethodName, ViewComponentTypes.SyncMethodName));
+                throw new InvalidOperationException(ViewComponentResources.FormatViewComponent_AmbiguousMethods(
+                    componentType.ToDisplayString(FullNameTypeDisplayFormat),
+                    ViewComponentTypes.AsyncMethodName,
+                    ViewComponentTypes.SyncMethodName));
```

Comment:
I removed the idea of **global errors** for taghelper discovery. So this exception is now just an exception. It sounds to me like this should be a diagnostic on the TagHelper.